drivers/src/nrfx_nvmc.c

```diff
@@ -306,15 +312,33 @@ bool nrfx_nvmc_page_partial_erase_continue(void)
 
 bool nrfx_nvmc_byte_writable_check(uint32_t addr, uint8_t val_to_check)
 {
-    NRFX_ASSERT((addr - NVMC_FLASH_BASE_ADDRESS) < flash_total_size_get());
+    NRFX_ASSERT(((addr - NVMC_FLASH_BASE_ADDRESS) < flash_total_size_get()) ||
+                ((addr - (uint32_t)NRF_UICR) < sizeof(NRF_UICR_Type)));
 
     uint8_t val_on_addr = *(uint8_t const *)addr;
     return (val_to_check & val_on_addr) == val_to_check;
 }
 
+bool nrfx_nvmc_halfword_writable_check(uint32_t addr, uint16_t val_to_check)
+{
+    NRFX_ASSERT(((addr - NVMC_FLASH_BASE_ADDRESS) < flash_total_size_get()) ||
+                ((addr - (uint32_t)NRF_UICR) < sizeof(NRF_UICR_Type)));
+    NRFX_ASSERT(is_halfword_aligned(addr));
+
+    uint16_t val_on_addr;
+
+    if ((addr - NVMC_FLASH_BASE_ADDRESS) < flash_total_size_get()) {
+        val_on_addr = *(uint16_t const *)addr;
+    } else {
```

Comment:
Move the opening brackets to next lines, and `else` as well, to align with nrfx coding style.

Reply:
Fixed